packages/go/ecosystem/ecosystem.go

```diff
@@ -0,0 +1,66 @@
+package ecosystem
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/ossf/osv-schema/packages/go/constants"
+)
+
+// Parsed represents an ecosystem-with-suffix string as defined by the [spec], parsed into
+// a structured format.
+//
+// The suffix is optional and is separated from the ecosystem by a colon.
+//
+// For example, "Debian:7" would be parsed into Parsed{Ecosystem: constants.EcosystemDebian, Suffix: "7"}
+//
+// [spec]: https://ossf.github.io/osv-schema/
```

Comment:
I think this can be more specifically pointing at https://ossf.github.io/osv-schema/#affectedpackage-field

Reply:
I considered that, but it didn't feel right as this isn't about the `affected` field itself, it's about the "ecosystem" type/value which happens to be used by the `affected` field - I really think ultimately the ecosystems table should have its own dedicated header as it's a very important part of the doc yet you currently can't directly link to it

Reply:
I don't see issue making the ecosystems table to have its own link. @oliverchang what do you think?

Reply:
yeah, I also got a few requests from others to make it easier to link to the ecosystems table

Reply:
I'll have a go at a PR for that then

Reply:
https://github.com/ossf/osv-schema/pull/306